plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -246,7 +250,7 @@ public IcebergTableHandle getTableHandle(ConnectorSession session, SchemaTableNa
         try {
             table = catalog.loadTable(session, new SchemaTableName(tableName.getSchemaName(), name.getTableName()));
         }
-        catch (TableNotFoundException e) {
+        catch (TableNotFoundException | UnknownTableTypeException e) {
```

Comment:
Add a test similar to https://github.com/trinodb/trino/commit/12a6b5f0b52cf80d42f00a18c077ca91342cdb83#diff-92c37834248d69878a49876bfa440c9f6afbee04b1363d402a04cf7fddc7bb62R2949-R2996 for checking that the querying of metadata tables for non-Iceberg tables is disallowed.

Reply:
This is already done in `io.trino.tests.product.iceberg.TestIcebergHiveTablesCompatibility`